### PR TITLE
Use mkdirp in fs adapter

### LINKED ISSUE #58: Fix error when the results folder doesn't exist
## Need

```gherkin
As a user
I want Mugshot to work correctly straight out of the box
So that I can quickly run my tests.
```

## Problem

```gherkin
Given that there is no `visual-tests` folder
When I run mugshot
Then it will throw with an error for each fixture telling me that there is no such file or directory.
```
Eg:
`Error: ENOENT: no such file or directory, mkdir 'tests/visual/visual-test/tooltip'`

## Deliverables

Fix the problem.

## Solution

Use [mkdirp](https://www.npmjs.com/package/mkdirp) instead of `fs.mkdir`.

### PR DESCRIPTION
**Pull request by adela-istrate-hs**
Closes #58

### CODE REVIEW

File: lib/adapters/fs.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ module.exports = objectAssign({}, FS, {
 
   writeFile: fs.writeFile,
 
-  mkdir: fs.mkdir,
+  mkdir: mkdirp,
```
Comment (catalin-miron-hs) [inline lib/adapters/fs.js]: can you add [fs-extra](https://www.npmjs.com/package/fs-extra) and use [mkdirp](https://github.com/jprichardson/node-fs-extra#mkdirsdir-callback). fs-extra it's a great lib and we should use is against regular 'fs'

Comment (andrei-picus-hs) [inline lib/adapters/fs.js]: @catalin-miron-hs what are the advantages over the current approach?

Comment (adela-istrate-hs) [inline lib/adapters/fs.js]: I don't know exacly the advantages but I will need `fs-extra` to delete the created folders in the integration test. So I think I should also use it in the lib to not have one extra package to install ...?

Comment (andrei-picus-hs) [inline lib/adapters/fs.js]: There's nothing wrong with installing another package for dev. For production, the overhead it brings should be measured.

For creating files and folders in tests we can and should use a temporary file/dir generator such as [tmp](https://www.npmjs.com/package/tmp).

Comment (adela-istrate-hs) [inline lib/adapters/fs.js]: Ok. I created an issue #69 to change the integration tests to use `tmp` and to add an integration test with a recursive `rootDirectory`. For now, can I merge this? @valentin-radulescu-hs

Comment (valentin-radulescu-hs) [inline lib/adapters/fs.js]: Go for it!

Comment (adela-istrate-hs) [inline lib/adapters/fs.js]: But Mugshot creates the files, so how can I do them temporary?

Comment (andrei-picus-hs) [inline lib/adapters/fs.js]: That's Mugshot's business. Your business is providing a temporary dir to store the results in. You can create a temp dir and set the `rootDirectory` option to a non-existent folder inside the temp one. You can then remove the entire temp dir at the end of the tests.

**Pull request merged**